editor/gui/editor_title_bar.cpp

```diff
@@ -40,7 +40,7 @@ void EditorTitleBar::gui_input(const Ref<InputEvent> &p_event) {
 		if (mm->get_button_mask().has_flag(MouseButtonMask::LEFT)) {
 			Window *w = Object::cast_to<Window>(get_viewport());
 			if (w) {
-				Point2 mouse = DisplayServer::get_singleton()->mouse_get_position();
```

Comment:
In this case, it should be global mouse position. EditorTitleBar is a custom editor window title bar used to move window, local coordinates will always go out of sync when window moves. It's used exclusively on macOS.

Reply:
`Viewport::get_embedder_mouse_position()` returns for a non-embedded window the screen coordinates and not the viewport local coordinates.

I have verified, that this change works on Linux by setting `EditorTitleBar::can_move = true`.